ceno_emul/tests/test_elf.rs

```diff
@@ -72,6 +74,75 @@ fn test_ceno_rt_io() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_ceno_rt_keccak() -> Result<()> {
+    let program_elf = ceno_examples::ceno_rt_keccak;
+    let mut state = VMState::new_from_elf(unsafe_platform(), program_elf)?;
+    let steps = run(&mut state)?;
+
+    // Expect the program to have written successive states between Keccak permutations.
+    const ITERATIONS: usize = 3;
+    let keccak_outs = sample_keccak_f(ITERATIONS);
+
+    let all_messages = read_all_messages(&state);
+    assert_eq!(all_messages.len(), ITERATIONS);
+    for (got, expect) in izip!(&all_messages, &keccak_outs) {
+        let got = got
+            .chunks_exact(8)
+            .map(|chunk| u64::from_le_bytes(chunk.try_into().unwrap()))
+            .collect_vec();
+        assert_eq!(&got, expect);
+    }
+
+    // Find the syscall records.
+    let syscalls = steps.iter().filter_map(|step| step.syscall()).collect_vec();
+    assert_eq!(syscalls.len(), ITERATIONS);
+
+    // Check the syscall effects.
+    for (witness, expect) in izip!(syscalls, keccak_outs) {
+        assert_eq!(witness.reg_accesses.len(), 1);
+        assert_eq!(
+            witness.reg_accesses[0].register_index(),
+            Platform::reg_arg0()
+        );
+
+        assert_eq!(witness.mem_writes.len(), expect.len() * 2);
+        let got = witness
+            .mem_writes
+            .chunks_exact(2)
+            .map(|write_ops| {
+                assert_eq!(
+                    write_ops[1].addr.baddr(),
+                    write_ops[0].addr.baddr() + WORD_SIZE as u32
+                );
+                let lo = write_ops[0].value.after as u64;
+                let hi = write_ops[1].value.after as u64;
+                lo | (hi << 32)
```

Comment:
We are doing the conversion from two `u32` to one one `u64` and back quite a few times in our code base.  Perhaps we should write to helper functions?

Reply:
Simple enough for now. Better do all such places at once some time.